lineapy/plugins/session_writers.py

```diff
@@ -0,0 +1,217 @@
+        )
+
+    def get_session_input_parameters_spec(
+        self, session_artifact
+    ) -> Dict[str, InputVariable]:
+        """
+        Return a dictionary with input parameters as key and InputVariable
+        class as value to generate code related to user input variables.
+        """
+        session_input_variables: Dict[str, InputVariable] = dict()
+        for line in self._get_session_input_parameters_lines(
+            session_artifact
+        ).split("\n"):
+            variable_def = line.strip(" ").rstrip(",")
+            if len(variable_def) > 0:
+                variable_name = variable_def.split("=")[0].strip(" ")
+                value = eval(variable_def.split("=")[1].strip(" "))
+                value_type = type(value)
+                if not (
+                    isinstance(value, int)
+                    or isinstance(value, float)
+                    or isinstance(value, str)
+                    or isinstance(value, bool)
+                ):
+                    raise ValueError(
+                        f"LineaPy only supports primitive types as input parameters for now. "
+                        f"{variable_name} in {variable_def} is a {value_type}."
+                    )
+                if ":" in variable_name:
+                    variable_name = variable_def.split(":")[0]
+                    session_input_variables[variable_name] = InputVariable(
+                        variable_name=variable_name,
+                        value=value,
+                        value_type=value_type,
+                    )
+                else:
+                    session_input_variables[variable_name] = InputVariable(
+                        variable_name=variable_name,
+                        value=value,
+                        value_type=value_type,
+                    )
+        return session_input_variables
+
+    def get_session_function_callblock(
+        self, session_artifact: SessionArtifacts
+    ) -> str:
+        """
+        Return the code to make the call to the session function as
+        `session_function_name(input_parameters)`.
+        """
+        session_function_name = self.get_session_function_name(
+            session_artifact
+        )
+        if session_function_name != "":
+            session_input_parameters = ", ".join(
+                self.get_session_input_parameters_spec(session_artifact).keys()
+            )
+            return f"{session_function_name}({session_input_parameters})"
+        else:
+            return ""
+
+    def get_session_artifact_function_definitions(
+        self, session_artifact: SessionArtifacts, indentation=4
+    ) -> List[str]:
+        """
+        Return the definition of each targeted artifacts calculation
+        functions.
+        """
+        return [
+            coll.get_function_definition(indentation=indentation)
+            for coll in session_artifact.artifact_nodecollections
+        ]
```

Comment:
Need to change if we change the variable name `artifact_nodecollections` in `session_artifacts.py`

Reply:
of course, need to resolve the merge conflict

Reply:
renamed, merge conflicts should be resolved too